WeakAuras/Prototypes.lua

```diff
@@ -11244,9 +11288,160 @@ Private.event_prototypes = {
+                currencyInfo.accountQuantity = currencyInfo.accountQuantity + currencyData.quantity
+              end
+            end
+          else
+            C_CurrencyInfo.RequestCurrencyDataForAccountCharacters()
+          end
+        end
+
+        local active = true
+        local primaryCheckFailed = false
+
+        -- Check quantity-related values
+        for attribute, triggerValue in pairs(triggerQuantities) do
+          if triggerValue ~= "nil" and not check(currencyInfo[attribute] or 0, triggerValue, triggerOperators[attribute]) then
+            active = false
+            if attribute ~= "quantity" and attribute ~= "realCharacterQuantity" then
+              primaryCheckFailed = true
+              break;
+            end
+          end
+        end
+
+        -- Helper function to check tristate values
+        local function checkTristate(tristateValue, attribute, condition, primary)
+          if active and tristateValue ~= "nil" and condition then
+            if tostring(attribute) ~= tristateValue then
+              active = false
+              primaryCheckFailed = primary and true
+            end
+          end
+        end
+
+        -- Check tristate values
+        checkTristate(discovered, currencyInfo.discovered, true, true)
+        checkTristate(capped, currencyInfo.quantity >= (currencyInfo.maxQuantity or 0), currencyInfo.maxQuantity and currencyInfo.maxQuantity > 0, false)
+        checkTristate(seasonCapped, currencyInfo.totalEarned >= (currencyInfo.maxQuantity or 0), currencyInfo.useTotalEarnedForMaxQty and currencyInfo.maxQuantity and currencyInfo.maxQuantity > 0, true)
+        checkTristate(weeklyCapped, currencyInfo.quantityEarnedThisWeek >= (currencyInfo.maxWeeklyQuantity or 0), currencyInfo.maxWeeklyQuantity and currencyInfo.maxWeeklyQuantity > 0, true)
+
+        local sharedStateValues = {
+          name = currencyInfo.name,
+          icon = currencyInfo.iconFileID,
+          total = currencyInfo.maxQuantity,
+          maxQuantity = currencyInfo.maxQuantity,
+          progressType = "static",
+          accountQuantity = currencyInfo.accountQuantity,
+          realAccountQuantity = currencyInfo.realAccountQuantity,
+          description = currencyInfo.description,
+          quality = currencyInfo.quality,
+          discovered = currencyInfo.discovered,
+          transferPercentage = currencyInfo.transferPercentage,
+          quantityEarnedThisWeek = currencyInfo.quantityEarnedThisWeek,
+          totalEarned = currencyInfo.totalEarned,
+          changed = true,
+        }
+
+        states[""] = states[""] or {}
+        local state = states[""]
+        state.show = active
+        state.value = currencyInfo.quantity
+        state.realCharacterQuantity = currencyInfo.realCharacterQuantity
+        state.capped = currencyInfo.maxQuantity and currencyInfo.maxQuantity > 0 and currencyInfo.quantity >= currencyInfo.maxQuantity
```

Comment:
These capped calculation is duplicated above in the checkTriState calls, the code above would be much easier to read if you calculated it just once.